hedera-node/test-clients/src/main/java/com/hedera/services/bdd/junit/SubProcessHapiTestNode.java

```diff
@@ -105,28 +184,27 @@ public void start() {
     }
 
     @Override
-    public void waitForActive(long seconds) {
+    public void waitForActive(long seconds) throws TimeoutException {
         final var waitUntil = System.currentTimeMillis() + (seconds * 1000);
-        final var log = workingDir.resolve("output").resolve("hgcaa.log");
         while (handle != null) {
             if (System.currentTimeMillis() > waitUntil) {
-                throw new RuntimeException(
+                throw new TimeoutException(
                         "node " + nodeId + ": Waited " + seconds + " seconds, but node did not become active!");
             }
 
-            try {
-                if (Files.exists(log)) {
-                    try (final var in = Files.newBufferedReader(log)) {
-                        String line;
-                        while ((line = in.readLine()) != null) {
-                            if (line.contains("ACTIVE")) {
-                                return;
-                            }
-                        }
-                    }
+            if ("ACTIVE".equals(getPlatformStatus())) {
+                // Actually try to open a connection with the node, to make sure it is really up and running.
+                // The platform may be active, but the node may not be listening.
+                try {
+                    final var url = new URL("http://localhost:" + grpcPort + "/");
+                    final var connection = url.openConnection();
+                    connection.connect();
+                    return;
+                } catch (MalformedURLException e) {
+                    throw new RuntimeException("Should never happen", e);
+                } catch (IOException ignored) {
+                    // This is expected, the node is not up yet.
                 }
-            } catch (Exception e) {
-                throw new RuntimeException("node " + nodeId + ": Unable to read from the hgcaa log file " + log, e);
             }
 
```

Comment:
Use the prometheus endpoint instead of scraping logs to make this more reliable.